src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -1277,64 +1277,72 @@ private static unsafe int ApplyMask(Span<byte> toMask, int mask, int maskIndex)
         {
             Debug.Assert(maskIndex < sizeof(int));
 
-            int maskShift = maskIndex * 8;
-            int shiftedMask = (int)(((uint)mask >> maskShift) | ((uint)mask << (32 - maskShift)));
-
-            // Try to use SIMD.  We can if the number of bytes we're trying to mask is at least as much
-            // as the width of a vector and if the width is an even multiple of the mask.
-            if (Vector.IsHardwareAccelerated &&
-                Vector<byte>.Count % sizeof(int) == 0 &&
-                toMask.Length >= Vector<byte>.Count)
-            {
-                Vector<byte> maskVector = Vector.AsVectorByte(new Vector<int>(shiftedMask));
-                Span<Vector<byte>> toMaskVector = MemoryMarshal.Cast<byte, Vector<byte>>(toMask);
-                for (int i = 0; i < toMaskVector.Length; i++)
+            fixed (byte* toMaskBeg = toMask)
+            {
+                byte* toMaskPtr = toMaskBeg;
+                byte* toMaskEnd = toMaskBeg + toMask.Length;
+                byte* maskPtr = (byte*)&mask;
+
+                if (toMaskEnd - toMaskPtr < sizeof(int))
                 {
-                    toMaskVector[i] ^= maskVector;
+                    goto small;
```

Comment:
Can we avoid the goto here?  Seems like you could just invert the condition and have everything after this be in the block.  This looks like it's the only jump to that label, and we generally prefer not to use goto unless it's demonstrated to be perf critical.